analyzers/golang/analyze.go

```diff
@@ -107,79 +107,94 @@ func (a *Analyzer) Analyze() (graph.Deps, error) {
 
 	log.Debugf("Resolver: %#v", r)
 
-	// Use `go list` to get imports and deps of module.
-	main, err := a.Go.ListOne(m.BuildTarget)
-	if err != nil {
-		return graph.Deps{}, err
-	}
-	log.Debugf("Go main package: %#v", main)
-	deps, err := a.Go.List(main.Deps)
-	if err != nil {
-		return graph.Deps{}, err
-	}
-
-	// Construct map of import path to package.
-	gopkgs := append(deps, main)
-	gopkgMap := make(map[string]gocmd.Package)
-	for _, p := range gopkgs {
-		gopkgMap[p.ImportPath] = p
-	}
-	// cgo imports don't have revisions.
-	gopkgMap["C"] = gocmd.Package{
-		Name:     "C",
-		IsStdLib: true, // This is so we don't try to lookup a revision. Maybe there should be a NoRevision bool field?
-	}
-	log.Debugf("gopkgMap: %#v", gopkgMap)
-
-	// Construct transitive dependency graph.
-	pkgs := make(map[pkg.ID]pkg.Package)
-	for _, gopkg := range deps {
-		log.Debugf("Getting revision for: %#v", gopkg)
-
-		// Resolve dependency.
-		revision, err := a.Revision(project, r, gopkg)
+	var allImports []pkg.Import
+	importMap := make(map[pkg.Import]bool)
+	transitiveDeps := make(map[pkg.ID]pkg.Package)
+
+	for _, buildTag := range a.BuildTags {
```

Comment:
Does this work for combination build tags? For example, let's say someone sets:

```
options:
  tags:
    - foo bar
```

Then `buildTag` will be `"foo bar"`. Can we pass that to `Go.ListOne`, or do we need to `strings.Fields` to pass `"foo"` and `"bar"` as separate arguments?

Reply:
Added a test for this but for your reference per `go help build`, `-tags` looks for `"tag1 tag2"`:
```
-tags 'tag list'
a space-separated list of build tags to consider satisfied during the
build. For more information about build tags, see the description of
build constraints in the documentation for the go/build package.```